app/components/swiper.js

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    InteractionManager,
+    View,
+    ScrollView,
+    ViewPagerAndroid,
+    Platform,
+    StyleSheet
+} from 'react-native';
+
+export default class Swiper extends PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        style: PropTypes.object,
+        showsPagination: PropTypes.bool,
+        paginationStyle: PropTypes.object,
+        initialPage: PropTypes.number,
+        width: PropTypes.number,
+        dotColor: PropTypes.string,
+        activeDotColor: PropTypes.string,
+        keyboardShouldPersistTaps: PropTypes.string,
+        onIndexChanged: PropTypes.func
+    };
+
+    static defaultProps = {
+        removeClippedSubviews: true, // add this to the scrollview
+        showsPagination: true,
+        initialPage: 0,
+        keyboardShouldPersistTaps: 'handled',
+        onIndexChanged: () => null
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.runOnLayout = true;
+        this.state = this.initialState(props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // if (this.props.width !== nextProps.width) {
+        //     if (nextProps.width > this.props.width) {
+        //         this.scrollToStart();
+        //     } else {
+        //         InteractionManager.runAfterInteractions(() => {
+        //             this.runOnLayout = true;
+        //         });
+        //     }
+        // }
+        this.setState(this.initialState(nextProps));
+    }
+
+    componentWillUpdate(nextProps, nextState) {
+        // If the index has changed, we notify the parent via the onIndexChanged callback
+        if (this.state.index !== nextState.index) {
+            this.props.onIndexChanged(nextState.index);
```

Comment:
This won't trigger the props to change, will it? Otherwise, we could get into a loop here

Reply:
it won't trigger the props to change